Project-Euler/Project-Euler/Program.cs

```diff
@@ -11,28 +11,93 @@ class Program
         static void Main()
         {
             PrintIntro();
-            PrintMenuOptions();//Eventually this method will not be called in main directly. This is for testing purposes.
+
+            while (true)
+            {
+                MenuOptions userSelection = GetMenuSelection();
+
+                if (userSelection == MenuOptions.A)
+                {
+                    DynamicProblemSelector.PrintAllProblems();
+                }
+                else if (userSelection == MenuOptions.S)
+                {
+                    DynamicProblemSelector.PrintProblem(GetProblemNumber(), typeof(Problems));
+                }
+                else // User Chose to Quit
+                {
+                    return;
+                }
+            }
         }
 
+        public enum MenuOptions { A, S, Q };
```

Comment:
Three suggestions:

- Can we make these Enums more descriptive? i.e. AllProblems, SelectProblem, Quit
- Since this enum will only ever be used in this class, it should be private.
- Put the enum in it's own space not hidden inbetween methods